src/retry-strategy.ts

```diff
@@ -1,7 +1,8 @@
 import { ErrorResponse, NetworkError } from './base-client';
 import { ConsoleLogger, EmptyLogger, Logger } from './logger';
+
 export interface RetryStrategy {
-    attemptRetry(error: Error): Promise<Error>;
+    attemptRetry(error: Error, ignoreRetryAfter?: boolean): Promise<Error>;
```

Comment:
Agree - I'm not super happy with this API yet - will probably turn it around anyway